src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -192,6 +196,14 @@ impl<'a> JobQueue<'a> {
         }
 
         if self.queue.is_empty() {
+            if !self.is_doc_all {
+                try!(writeln!(cx.config.shell().err(), "Finished {} build.",
+                    if self.is_release {
+                        "release (optimized)"
+                    } else {
+                        "debug (unoptimized + debuginfo)"
```

Comment:
It might be worth checking the default profile here to tweak how this is printed. I believe this can be calculated when the job queue is created by taking a look at [`Context::lib_profile`](https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_rustc/context.rs#L638) with the argument of `cx.resolve.root()`.  The [`Profile`](https://github.com/rust-lang/cargo/blob/master/src/cargo/core/manifest.rs#L122) can then tell us whether it's optimized (`opt_level > 0`) or if `debuginfo` is enabled.

That way we can basically correctly say:

```
Finished release (optimized + debuginfo) build
```